public/lib/xmlhttprequest.js

```diff
@@ -166,13 +166,19 @@
               }
 
           }
-    if(this._request.status > 399){
-      var err = {
-        statusText: this._request.statusText,
-        status: this._request.status
-      };
-      this.callbackError(err);
-      return;
+    //mobile devices(iPad have thrown invalid state error for this check)
+    try {
+      if (this._request.status > 399) {
```

Comment:
could we add a description for why 399 is chosen?
For status messages 4xx and above this is happens.?